### PR TITLE
refactor: adopt --app as celery global option

### PR DESCRIPTION
**Pull request by shawnzhu**
### SUMMARY
This is the [step 1](https://docs.celeryproject.org/en/stable/history/whatsnew-5.0.html#step-1-adjust-your-command-line-invocation) of upgrading to Celery 5

> Celery 5.0 introduces a new CLI implementation which isn’t completely backwards compatible.
The global options can no longer be positioned after the sub-command. Instead, they must be positioned as an option for the celery command like so:

    celery --app path.to.app worker

### ADDITIONAL INFORMATION

- [x] Has associated issue: #14925
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by shawnzhu** [comment]
I hope this PR will make review of #14926 (a downstream dependency) easier

**Review by nytai — APPROVED** [review]

**Comment by nytai** [comment]
@shawnzhu thanks for the contribution! Looks like this change is compatible with the current version of celery so happy to merge once CI passes. As for https://github.com/apache/superset/pull/14926, it will require more thorough testing of our celery dependant features as it is a significant change.

**Comment by shawnzhu** [comment]
@nytai Thanks for the feedback! Now i can get #14926 to pass CI after this is merged

**Pull request merged**
